test/built-ins/GeneratorPrototype/next/context-constructor-invocation.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 25.2
+description: >
+    When a generator function is invoked as a constructor, its context is the
+    newly-created iterable object.
+---*/
+
+var context;
+function* g() { context = this; }
```

Comment:
This caught me off guard. Doing my due diligence, I verified in the spec:

> **9.2.3 FunctionAllocate (functionPrototype, strict [,functionKind] )**
> ...
> 9. If needsConstruct is true, then
>    a. Set F’s [[Construct]] internal method to the definition specified in
>       9.2.2.
>    b. If functionKind is "generator", set the [[ConstructorKind]] internal
>       slot of F to "derived".
>    c. Else, set the [[ConstructorKind]] internal slot of F to "base".
>    d. NOTE Generator functions are tagged as "derived" constructors to
>       prevent [[Construct]] from preallocating a generator instance.
>       Generator instance objects are allocated when EvaluateBody is applied
>       to the GeneratorBody of a generator function.

and:

> **9.2.2 `[[Construct]]` ( argumentsList, newTarget)**
> ...
> 5. If kind is "base", then
>    a. Let thisArgument be OrdinaryCreateFromConstructor(newTarget,
>       "%ObjectPrototype%").
>    b. ReturnIfAbrupt(thisArgument).

Thanks!